baseplate/clients/thrift.py

```diff
@@ -223,119 +260,150 @@ def _call_thrift_method(self: Any, *args: Any, **kwargs: Any) -> Any:
+
+                            min_timeout = time_remaining
+                            if self.pool.timeout:
+                                if not min_timeout or self.pool.timeout < min_timeout:
+                                    min_timeout = self.pool.timeout
+                            if min_timeout and min_timeout > 0:
+                                # min_timeout is in float seconds, we are converting to int milliseconds
+                                # rounding up here.
+                                prot.trans.set_header(
+                                    b"Deadline-Budget", str(int(ceil(min_timeout * 1000))).encode()
+                                )
+
+                            try:
+                                edge_context = span.context.raw_edge_context
+                            except AttributeError:
+                                edge_context = None
+
+                            if edge_context:
+                                prot.trans.set_header(b"Edge-Request", edge_context)
+
+                            result = method(*args, **kwargs)
+                        except TTransportException as exc:
+                            # the connection failed for some reason, retry if able
                             span.finish(exc_info=sys.exc_info())
+                            otelspan.set_status(status.Status(status.StatusCode.ERROR))
+                            last_error = str(exc)
+                            if exc.inner is not None:
+                                last_error += f" ({exc.inner})"
+                            raise  # we need to raise all exceptions so that self.pool.connect() self-heals
+                        except (TApplicationException, TProtocolException):
+                            # these are subclasses of TException but aren't ones that
+                            # should be expected in the protocol. this is an error!
+                            span.finish(exc_info=sys.exc_info())
+                            otelspan.set_status(status.Status(status.StatusCode.ERROR))
+                            raise
+                        except Error as exc:
+                            # a 5xx error is an unexpected exception but not 5xx are
+                            # not.
+                            if 500 <= exc.code < 600:
+                                span.finish(exc_info=sys.exc_info())
+                                otelspan.set_status(status.Status(status.StatusCode.ERROR))
+                            else:
+                                span.finish()
+                            raise
+                        except TException:
+                            # this is an expected exception, as defined in the IDL
+                            otelspan.set_status(status.Status(status.StatusCode.OK))
+                            span.finish()
+                            raise
+                        except:  # noqa: E722
```

Comment:
```suggestion
                        except BaseException:
```

Explicitly writing `except BaseException:` is the way to avoid the linter error when that's your goal.